script.artistslideshow/default.py

```diff
@@ -1075,11 +997,302 @@ def _upgrade( self ):
         #this is where any code goes for one time upgrade routines
         loglines, data = readFile( self.CHECKFILE )
         lw.log( loglines )
-        if not data:
-            self._migrate_info_files()
-        loglines, data = readFile( self.CHECKFILE )
-        if data == '1.5.4':
-            self._migrate_tbn_files()
+        if '2.1.0' not in data:
+            self._upgrade_artist_folders()
+            self._upgrade_artist_images()
+            self._update_check_file( self.CHECKFILE, '2.1.0', 'name change of artist folders and image files complete' )
+        loglines, upgradecheck = readFile( self.CHECKFILE )
+        lw.log( loglines )
+        loglines, imagecheck = readFile( self.IMAGECHECKFILE )
+        lw.log( loglines )
+        if not 'true' in imagecheck:
+            lw.log( ['upgradecheck is %s and localstorageonly is %s' % (upgradecheck, self.LOCALSTORAGEONLY)] )
+            if '2.1.0' in upgradecheck and self.LOCALSTORAGEONLY == 'true':
+                lw.log( ['migrating images'] )
+                self._upgrade_migratetolocal()
+                self._update_check_file( self.IMAGECHECKFILE, 'true', 'images migrated to local storage location' )
+
+        
+    def _upgrade_artist_folders( self ):
+        inforoot = os.path.join( self.DATAROOT, 'ArtistInformation' )
+        imgroot = os.path.join( self.DATAROOT, 'ArtistSlideshow' )
+        infoarchiveroot = os.path.join( self.DATAROOT, 'archive', 'ArtistInformation' )
+        imgarchiveroot = os.path.join( self.DATAROOT, 'archive', 'ArtistSlideshow' )
+        artist_hashes = self._upgrade_get_artists_hashmap()
+        aDialog = xbmcgui.DialogProgressBG()
+        aDialog.create( smartUTF8(language(32013)), smartUTF8(language(32012)) )
+        try:
+            info_dirs, old_files = xbmcvfs.listdir( inforoot )
+        except Exception, e:
```

Comment:
`except Exception as e:` is the preferred syntax as it is python3 compatible

Reply:
Thanks for the heads up.  I've been updating that as I update code that I actually run in Python3, but I hadn't checked to see if that syntax was backwards compatible.  Do I need to fix this for the current pull request, or can I just stick it in my todo list for the next update?

Reply:
To do list is fine